gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/VerificationOnboardService.java

```diff
@@ -74,58 +70,114 @@ public Optional<String> findSwaggerUrl(Map<String, String> metadata) {
      * @return Swagger as string
      */
     public String getSwagger(String swaggerUrl) {
-        String response;
-        response = restTemplate.getForEntity(swaggerUrl, String.class).getBody();
-        return response;
+        HttpHeaders headersSSO = new HttpHeaders();
+        headersSSO.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<Object> requestSSO = new HttpEntity<>(headersSSO);
+        return restTemplate.exchange(swaggerUrl, HttpMethod.GET, requestSSO, String.class).getBody();
     }
 
     /**
      * Checks if endpoints can be called and return documented responses
      *
-     * @param getEndpoints GET endpoints to check
+     * @param getEndpoints              GET endpoints to check
+     * @param passedAuthenticationToken Token used to call endpoints that support SSO
      * @return List of problems
      */
-    public List<String> testGetEndpoints(Set<Endpoint> getEndpoints) {
+    public List<String> testEndpointsByCalling(Set<Endpoint> getEndpoints, String passedAuthenticationToken) {
         ArrayList<String> result = new ArrayList<>();
 
+        String ssoCookie = getAuthenticationCookie(passedAuthenticationToken);
+
+        HttpHeaders headersSSO = new HttpHeaders();
+        headersSSO.setContentType(MediaType.APPLICATION_JSON);
+        headersSSO.add("Cookie", "apimlAuthenticationToken=" + ssoCookie);
+        HttpEntity<String> requestSSO = new HttpEntity<>(headersSSO);
+
+        HttpHeaders headersNoSSO = new HttpHeaders();
+        headersNoSSO.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<String> requestNoSSO = new HttpEntity<>(headersNoSSO);
+
         for (Endpoint endpoint : getEndpoints) {
-            String urlFromSwagger = endpoint.getUrl();
-            // replaces parameters in {} in query
-            String url = urlFromSwagger.replaceAll("\\{[^{}]*}", "dummy");
-
-            ResponseEntity<String> response;
-            try {
-                response = restTemplate.getForEntity(url, String.class);
-            } catch (HttpClientErrorException | HttpServerErrorException e) {
-                response = ResponseEntity.status(e.getRawStatusCode())
-                    .headers(e.getResponseHeaders())
-                    .body(e.getResponseBodyAsString());
+            for (HttpMethod method : endpoint.getHttpMethods()) {
+                checkEndpoint(endpoint, result, method, requestSSO, requestNoSSO);
             }
+        }
+        return result;
+    }
+
+    private void checkEndpoint(Endpoint endpoint, ArrayList<String> result, HttpMethod method, HttpEntity<String> requestSSO, HttpEntity<String> requestNoSSO) {
+        if (method == HttpMethod.DELETE) {
+            return;
+        }
 
-            String responseBody = response.getBody();
+        String urlFromSwagger = endpoint.getUrl();
+        // replaces parameters in {} in query
+        String url = urlFromSwagger.replaceAll("\\{[^{}]*}", "dummy");
```

Comment:
why dummy? it would be good for posterity to provide example, e.g. "original->result"

Reply:
Edited the comment to have an example + explanation